src/components/Modals/Send/hooks/useSendDetails/useSendDetails.test.tsx

```diff
@@ -199,19 +205,21 @@ xdescribe('useSendDetails', () => {
     // Set fiat amount
     await act(async () => {
       result.current.handleInputChange('3500')
-      await new Promise(r => setTimeout(r, 1500)) // hack to wait because handleInputChange is now debounced for 1 second
```

Comment:
this promise was triggering some network calls hence I replaced it

Reply:
This doesn't make sense to me. Waiting can't cause a network request by itself, it can only allow enough time for something else to try to make one. We should figure out what's causing the underlying behavior rather than just out-racing it.

Reply:
agreed with @reidrankin - i was never a fan of this sleep hack, and prefer to use jest advance timers, but we should at least advance by as much or more (because in real time it's free), and also agree this is probably masking the issue

Reply:
@chxash i can verify that with the current implementation tests do indeed run offline 🤘

Reply:
hi @reidrankin, thank you for your feedback, I agree with you the promise could not have generated network calls. I investigated a little bit more, with latest develop and promises => I am not able to reproduce the network calls I saw multiple times in that file.
@0xdef1cafe I fixed the timer advance to 1500, like it was before with promises.

Reply:
hi all,

I'd like to propose using msw (https://github.com/mswjs/msw)
I've been using it in many of my projects and I have had a very positive experience with it

it intercepts requests on the network level,
that would allow us properly isolate the tests without a need to jest.mock '@shapeshiftoss/market-service' (or any other service)

I've created a draft, please take a look: https://github.com/shapeshift/web/pull/807

the tests run offline and you can see a few errors like
```
[MSW] Error: captured a request without a matching request handler:
```

Reply:
we could use it to define API mocks later

with API mocks it wouldn't be necessary to mock slices/selectors
instead, you define API responses for the test cases that you need to cover 

it would make tests less brittle, prevent exposing implementation details in tests
what would enable refactoring and have a positive effect on quality in a long term

another pro: the same mock definition could be reused for E2E testing with cypress, debugging, or even developing in a browser 😄